Middleware/GuzzleRequestSpanning.php

```diff
@@ -33,6 +35,23 @@ function (ResponseInterface $response) {
                     $this->requestSpanning->finish($response->getStatusCode());
                     $this->tracing->finishActiveSpan();
                     return $response;
+                },
+                function (RequestException $e) {
+                    $this->tracing->setTagOfActiveSpan(ERROR, true);
+                    $this->tracing->logInActiveSpan(
+                        [
+                            'event' => 'error',
+                            'error.kind' => 'Exception',
+                            'error.object' => get_class($e),
+                            'message' => $e->getMessage(),
+                            'stack' => $e->getTraceAsString(),
+                        ]
+                    );
+                    if ($e->hasResponse()) {
+                        $this->requestSpanning->finish($e->getResponse()->getStatusCode());
+                    }
+                    $this->tracing->finishActiveSpan();
+                    throw $e;
```

Comment:
codeclimate:
> Method __invoke has 28 lines of code (exceeds 25 allowed). Consider refactoring. 

Maybe it is time to separate the callables into own methods within this class.

Reply:
done